# Worksheet05

This is for your joy, amusement, and edification. Don't think of it as an assignment. Think of it as a chance to exercise your skills. 

You may collaborate with friends, refer to notes, search the web, or use any other resources as you work on the problems below.

Relevant notes are from September 11th and 13th.

Run the following cell to set everything up.

In [1]:
%matplotlib inline
import numpy as np
import pylab

#Here's your data
np.random.seed(0) 
D = np.random.randint(-50, 50, (10, 5))
D 

array([[ -6,  -3,  14,  17,  17],
       [-41,  33, -29, -14,  37],
       [ 20,  38,  38, -38,   8],
       [ 15, -11,  37,  -4,  38],
       [ 31, -13, -25,  27,  22],
       [-41, -30,  30,  19,  29],
       [ -3,  14,  32,  49,  38],
       [ -1, -21, -31, -31, -36],
       [-11, -18,  15, -41,   7],
       [-18, -19,  24, -27, -15]])

1) Count how many entries are less than 0

In [2]:
# write your code here
np.sum(D<0)

25

I got 25

2) Are any of the entries in the the 4th column (the index being 3) divisible by 8?

In [3]:
D[:,3]
(D[:,3])/8
np.any((D[:,3])%8==0)

False

I got False, there are no entries in the 4th column divisible by 8

3) How many rows have all even numbers?

In [4]:
# write your code here
#np.sum(np.all(D/2))
np.count_nonzero(np.all(D%2==0, axis=1))

1

I got 1

4) What is the maximum value of each column?

In [5]:
# write your code here
np.max(D, axis=0)

array([31, 38, 38, 49, 38])

I got [31, 38, 38, 49, 38]

5) Reshape the array D into a new array called E with the shape (2,25), that is 2 rows and 25 columns

In [6]:
# write your code here
# E = 
E= np.reshape(D, (2,25))
E

array([[ -6,  -3,  14,  17,  17, -41,  33, -29, -14,  37,  20,  38,  38,
        -38,   8,  15, -11,  37,  -4,  38,  31, -13, -25,  27,  22],
       [-41, -30,  30,  19,  29,  -3,  14,  32,  49,  38,  -1, -21, -31,
        -31, -36, -11, -18,  15, -41,   7, -18, -19,  24, -27, -15]])

6) What's the mean of E?

In [7]:
# write your code here
np.mean(E)

2.4399999999999999

I got 2.4399999999999999

7) How many columns of E have elements that are all greater than the mean of E?

In [8]:
# write your code here
greater=E>np.mean(E)
columns=np.count_nonzero(greater,axis=0)
np.count_nonzero(columns==2)

7

I got 7

When you run the cell below you should get a **Submit this notebook** button. Click that button, login, and your notebook should be recorded.

In [9]:
# do not change the code below
import comp116

comp116.submit('ws5')